File: MAG/UI/mag_admin/lib/models/post.dart
import 'package:json_annotation/json_annotation.dart';
part 'post.g.dart';

@JsonSerializable()
class Post {
  int? id;
  int? clubId;
  int? userId;
  String? content;
  int? likesCount;
  int? dislikesCount;
  DateTime? datePosted;

  Post(
    this.id,
    this.clubId,
    this.userId,
    this.content,
    this.likesCount,
    this.dislikesCount,
    this.datePosted,
  );

  factory Post.fromJson(Map<String, dynamic> json) => _$PostFromJson(json);

  Map<String, dynamic> toJson() => _$PostToJson(this);
}

File: MAG/MAG.Services/Database/Post.cs
using System;
using System.Collections.Generic;

namespace MAG.Services.Database;

public partial class Post
{
    public int Id { get; set; }

    public int ClubId { get; set; }

    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public int LikesCount { get; set; }

    public int DislikesCount { get; set; }

    public DateTime DatePosted { get; set; }

    public virtual Club Club { get; set; } = null!;

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual User User { get; set; } = null!;

    public virtual ICollection<UserPostAction> UserPostActions { get; set; } = new List<UserPostAction>();
}
